src/vm/moar/ops/container.c

```diff
@@ -79,7 +79,10 @@ static void rakudo_scalar_store(MVMThreadContext *tc, MVMObject *cont, MVMObject
     if (rcd && IS_CONCRETE(rcd))
         rw = rcd->rw;
     if (!rw)
-        MVM_exception_throw_adhoc(tc, "Cannot assign to a readonly variable or a value");
+        if (rcd && IS_CONCRETE(rcd) && rcd->name)
+            MVM_exception_throw_adhoc(tc, "Cannot assign to a readonly variable (%s) or a value", MVM_string_utf8_encode_C_string(tc, rcd->name));
```

Comment:
This leaks the encoded C string; please use `MVM_exception_throw_adhoc_free`. Example use here: https://github.com/MoarVM/MoarVM/blob/bc6e933a0688354396fb1a1b97363375fd10a138/src/6model/reprs/MVMContext.c#L37